### ISSUE TITLE
[Bug]: Truncate a non-hypertable table on postgres with TimeScaleDB lead to "Segmentation fault" 

### ISSUE THREAD

**Issue by OferGuy** [timescale/timescaledb#3856]
### What type of bug is this?

Crash

### What subsystems and features are affected?

Query planner, Query executor, Command processing, Other

### What happened?

We have a java code running every 5min to truncate a table which is not a hypertable. 
In many cases, even table is empty, it will lead to "Segmentation fault" and crash the postgres DB: 

/var/log/xxxuserxxx/postgres_stderr.log
2021-11-24 13:29:08 UTC::@:[90]:LOG:  server process (PID 122655) was terminated by signal 11: Segmentation fault
2021-11-24 13:29:08 UTC::@:[90]:DETAIL:  Failed process was running: truncate table CEInterfaceUtilization15Min
2021-11-24 13:29:08 UTC::@:[90]:LOG:  terminating any other active server processes
2021-11-24 13:29:08 UTC:10.56.112.149(64613):xxxuserxxx@xxxdbxxx:[125340]:WARNING:  terminating connection because of crash of another server process

### TimescaleDB version affected

2.2.1

### PostgreSQL version used

"PostgreSQL 13.3 on x86_64-alpine-linux-musl, compiled by gcc (Alpine 10.2.1_pre1) 10.2.1 20201203, 64-bit"

### What operating system did you use?

Alpine Linux" ID=alpine VERSION_ID=3.13.0

### What installation method did you use?

Installing TimeScaleDB apk through a Docker file

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
/var/log/xxxuserxxx/postgres_stderr.log
2021-11-24 13:29:08 UTC::@:[90]:LOG:  server process (PID 122655) was terminated by signal 11: Segmentation fault
2021-11-24 13:29:08 UTC::@:[90]:DETAIL:  Failed process was running: truncate table CEInterfaceUtilization15Min
2021-11-24 13:29:08 UTC::@:[90]:LOG:  terminating any other active server processes
2021-11-24 13:29:08 UTC:10.56.112.149(64613):XXXUSERXXX@XXXDBXXX:[125340]:WARNING:  terminating connection because of crash of another server process


```

### How can we reproduce the bug?

```bash
it is easily reproduced.
```

**Comment by NunoFilipeSantos** [comment]
Hi @OferGuy ! 👋  Can you try please TimescaleDB 2.5.0 and get back to us? Thank you!

**Comment by Sasha-Velkovich** [comment]
Hi @NunoFilipeSantos , 
We tried to install TSDB 2.5.0 apk (over postgres 13.3)  - it complained that postgres14 is needed. Currently we can't upgrade to postgres14 . 
Here is the apk we tried to use : (there are requirement for postgres 14) 
https://alpine.pkgs.org/3.15/alpine-community-x86_64/postgresql-timescaledb-2.5.0-r1.apk.html

I found only TSDB 2.4 in the official repo : https://pkgs.alpinelinux.org/packages?name=postgresql-timescaledb&branch=edge&repo=testing

Do you suggest to build it from the sources ? will it support postgres13.3 ?

**Comment by svenklemm** [comment]
This should be fixed by #3626 in 2.5.0. Feel free to reopen if this still happens in 2.5.0+.
